lib/routes/routeBackbeat.js

```diff
@@ -464,6 +464,47 @@ function putData(request, response, bucketInfo, objMd, log, callback) {
     });
 }
 
+/**
+ * @callback CanonicalIdCallback
+ * @param {Error|null} err - Error object if operation failed, null otherwise
+ * @param {Object} [result] - Result object containing account information
+ * @param {string} [result.accountId] - The account ID
+ * @param {string} [result.canonicalId] - The canonical ID associated with the account
+ * @param {string} [result.name] - The display name associated with the account
+ * @returns {undefined}
+ */
+
+/**
+ * 
+ * @param {string} accountId - account ID
+ * @param {Log} log - logger instance
+ * @param {CanonicalIdCallback} cb - callback function
+ * @returns {undefined}
+ */
+function getCanonicalIdsByAccountId(accountId, log, cb) {
+    if (!accountId) {
+        return cb();
+    }
+    return vault.getCanonicalIdsByAccountIds([accountId], log, (err, res) => {
+        if (err) {
+            log.error('error getting canonical ID by account ID', {
+                error: err,
+                accountId,
+                method: 'getCanonicalIdsByAccountIds',
+            });
+            return cb(err);
+        }
+        if (res.length === 0) {
+            log.error('account ID not found', {
+                accountId,
+                method: 'getCanonicalIdsByAccountIds',
+            });
+            return cb(errors.AccountNotFound);
+        }
+        return cb(null, res[0]);
```

Comment:
To avoid having to deal with arrays, Vault supports the `getAccount` API. This API accepts any info to uniquely identity an account, including the name, email, and canonical id.
See https://github.com/scality/vault2/blob/development/8.10/lib/Handlers/Account.js#L1041

If we only use one account at a time, preefer using this API. This will also simplify handling in Arsenal.